lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -26,4 +28,392 @@ using namespace mlir::torch::onnx_c;
+                    return failure();
+                  rewriter.replaceOpWithNewOp<Torch::AtenSinOp>(
+                      binder.op, resultType, operand);
+                  return success();
+                });
+  patterns.onOp("Tanh", 1,
+                [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+                  Torch::ValueTensorType resultType;
+                  Value operand;
+                  if (binder.tensorOperand(operand) ||
+                      binder.tensorResultType(resultType))
+                    return failure();
+                  rewriter.replaceOpWithNewOp<Torch::AtenTanhOp>(
+                      binder.op, resultType, operand);
+                  return success();
+                });
+  patterns.onOp("Sqrt", 1,
+                [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+                  Torch::ValueTensorType resultType;
+                  Value operand;
+                  if (binder.tensorOperand(operand) ||
+                      binder.tensorResultType(resultType))
+                    return failure();
+                  rewriter.replaceOpWithNewOp<Torch::AtenSqrtOp>(
+                      binder.op, resultType, operand);
+                  return success();
+                });
+  patterns.onOp(
+      "Sub", 1, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value x;
+        Value y;
+        if (binder.tensorOperands(x, y) || binder.tensorResultType(resultType))
+          return failure();
+        Value const1 = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getIntegerAttr(rewriter.getIntegerType(64), 1));
+        rewriter.replaceOpWithNewOp<Torch::AtenSubTensorOp>(
+            binder.op, resultType, x, y, const1);
+        return success();
+      });
+  patterns.onOp(
+      "Sum", 1, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        if (binder.op->getNumOperands() == 1) {
+          Torch::ValueTensorType resultType;
+          Value x;
+          if (binder.tensorOperand(x) || binder.tensorResultType(resultType))
+            return failure();
+          rewriter.replaceOp(binder.op, x);
+          return success();
+        }
+        if (binder.op->getNumOperands() == 2) {
+          Torch::ValueTensorType resultType;
+          Value x;
+          Value y;
+          if (binder.tensorOperands(x, y) ||
+              binder.tensorResultType(resultType))
+            return failure();
+          Value const1 = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(),
+              rewriter.getIntegerAttr(rewriter.getIntegerType(64), 1));
+          rewriter.replaceOpWithNewOp<Torch::AtenAddTensorOp>(
+              binder.op, resultType, x, y, const1);
+          return success();
+        }
+        // When binder.op->getNumOperands() > 2
```

Comment:
I'm not sure this choice of how to translate is the best idea: you are introducing a potentially very expensive data movement step to compute the sun, which is not implied by the original op def.

I'd recommend just producing n-1 add ops. Do your setup for the first two but don't return. Then process 2..n but adding to the last result and return the last result at the end. One exit, no data movement.

Reply:
I thought this was a neat way to lower, but this makes total sense. I will definitely consider data movement costs going forward. Made necessary changes